dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/tag/TagRouter.java

```diff
@@ -199,11 +201,26 @@ private boolean isForceUseTag(Invocation invocation) {
     }
 
     private boolean addressMatches(URL url, List<String> addresses) {
-        return addresses != null && addresses.contains(url.getAddress());
+        return addresses != null && checkAddressMatch(addresses, url.getAddress());
     }
 
     private boolean addressNotMatches(URL url, List<String> addresses) {
-        return addresses == null || !addresses.contains(url.getAddress());
+        return addresses == null || !checkAddressMatch(addresses, url.getAddress());
+    }
+
+    private boolean checkAddressMatch(List<String> addresses, String targetAddress) {
+        for (String address : addresses) {
+            try {
+                if (NetUtils.matchIpExpression(address, targetAddress)) {
```

Comment:
The address here is expected to have port included: `127.0.0.1:20880`, but `NetUtils.matchIpExpression` can only handle raw IP.

Reply:
I have fixed this one to support port configuration.